File: Week7/cat-app/src/app/services/cat.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { Cat } from '../models/cat';
import { UrlService } from './url.service';
import { Person } from '../models/person';
import { Breed } from '../models/breed';
import { SpecialNeed } from '../models/special-need';

@Injectable({
  providedIn: 'root'
})
export class CatService {
  private url: string = this.urlServ.getUrl() + 'cats/';

  constructor(private urlServ: UrlService, private http: HttpClient) { }

  getAvailableCats(): Observable<Cat[]> {
    return this.http.get(this.url, {withCredentials:true}).pipe(
      map(resp => resp as Cat[])
    );
  }

  getCatById(id: number): Observable<Cat> {
    return this.http.get(this.url + id, {withCredentials:true}).pipe(
      map(resp => resp as Cat)
    );
  }

  adoptCat(cat: Cat): Observable<any> {
    return this.http.get(this.url + 'adopt/' + cat.id, {withCredentials:true}).pipe();
  }

  getBreeds(): Observable<Breed[]> {
    return this.http.get(this.url + 'breeds/', {withCredentials:true}).pipe(
      map(
        resp => resp as Breed[]
      )
    );
  }

  getSpecialNeeds(): Observable<SpecialNeed[]> {
    return this.http.get(this.url + 'specialneeds/', {withCredentials:true}).pipe(
      map( resp => resp as SpecialNeed[] )
    );
  }
}

File: extras/MockitoSpringTest/CatAppBoot/src/main/java/com/revature/app/services/CatService.java
package com.revature.app.services;

import java.util.Set;

import com.revature.app.beans.Breed;
import com.revature.app.beans.Cat;
import com.revature.app.beans.Person;

public interface CatService {
	public Integer addCat(Cat c);
	public Cat getCatById(Integer id);
	public Set<Cat> getCats();
	public Set<Cat> getAvailableCats();
	public Set<Breed> getBreeds();
	public void updateCat(Cat c);
	public void adoptCat(Person p, Cat c);
	public void removeCat(Cat c);

}
